includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1544,14 +1501,30 @@ protected function build_display_items( $itemized_display_items = false ) {
 			];
 		}
 
-		return [
-			'displayItems' => $items,
-			'total'        => [
-				'label'   => $this->total_label,
-				'amount'  => max( 0, apply_filters( 'woocommerce_stripe_calculated_total', WC_Stripe_Helper::get_stripe_amount( $order_total ), $order_total, WC()->cart ) ),
-				'pending' => false,
-			],
+		// Order total
+		if ( version_compare( WC_VERSION, '3.2', '<' ) ) {
+			$items_total = wc_format_decimal( WC()->cart->cart_contents_total, WC()->cart->dp ) + $discounts;
+			$order_total = wc_format_decimal( $items_total + $tax + $shipping - $discounts, WC()->cart->dp );
```

Comment:
Another comment not related to your changes specifically, but more as a sanity check:

- [ ] If we're adding `$discounts` to `$items_total` and subtracting it from `$items_total` again in `$order_total`, couldn't we simply not add and subtract it at all?

Reply:
Good observation. This is very old code. It's been like that since 4 years ago https://github.com/woocommerce/woocommerce-gateway-stripe/commit/dad8dc3af52a3f01a03050a72950610f50304672

I haven't tested extensively with discounts but it seems weird discounts are not subtracted to calculate totals unless it was already considered in `WC()->cart->cart_contents_total`.

Not sure how to test this since it applies only for WCStripe v3.2.

Reply:
done e7e17c3